docs/site/Decorators.md

```diff
@@ -132,56 +132,183 @@ accordingly or do a composition of them:
-  decorator must be applied after an operation decorator.
-  To learn more about TypeScript decorator composition,
-  refer to [TypeScript Decorator Documentation](https://www.typescriptlang.org/docs/handbook/decorators.html)
+  You can find the specific usage in [Writing Controller methods](controllers.md#writing-controller-methods)
 
-  Please note method level `@param` and parameter level `@param` are mutually exclusive,
-  you can not mix and apply them to the same parameter.
+  *The parameter location cookie is not supported yet, see*
+  *https://github.com/strongloop/loopback-next/issues/997*
 
-  You can also use this pattern to make it quicker to define params: `@param.${in}.${type}(${name})`
+### RequestBody Decorator
 
-  - in: one of the following values: `query`, `header`, `path`, `formData`, `body`
-  - type: one of the following values:  `string`, `number`, `boolean`, `integer`
-  - name: a `string`, name of the parameter
+  Syntax: see [API documentation](https://github.com/strongloop/loopback-next/blob/master/packages/openapi-v3/src/request-body-decorator.ts#L20-L79)
+
+  `@requestBody()` is applied to a controller method parameter to generate OpenAPI requestBody specification for it.
+  
+  *Only one parameter can be decorated by `@requestBody` per controller method.*
+
+  A typical OpenAPI requestBody spec contains properties `description`, `required`, and `content`:
+
+  ```ts
+    requestBodySpec: {
+    description: 'a user',
+    required: true,
+    content: {
+      'application/json': {...schemaSpec},
+      'application/text': {...schemaSpec},
+    },
+  }
+  ```
+
+  When the decorated argument is a custom type or a model, we recommend you define 
+  a model class with `@model` and `@property`:
+  
+  ```ts
+  import {model, property} from '@loopback/repository';
+  import {Address} from './address.model';
+
+  @model()
+  class User {
+    @property()
+    firstname: string
+    @property()
+    lastname: string
+    @property()
+    address: Address
+  }
+  ```
+  *To learn more about decorating models and the corresponding OpenAPI schema, please check*
+  *[model decorators](#model-decorators).*
+
+  Then a typical usage of `@requestBody` is specifying a parameter's type as the model class,
+  and applying the decorator on it:
+
+  ```ts
+  import {User} from '../models/user'
+  import {put} from '@loopback/rest'
+  // in file controller.ts
+  class UserController {
+    @put('/Users/{id}')
+    async replaceUser(
+      @param.path.string('id') id: string,
+      @requestBody() user: User
+    ) {}
+  }
+  ```
+
+  For the simplest use case, you can leave the input of `@requestBody` empty, 
+  since we automatically detect the type of `user`, and generate the corresponding schema for it. 
+  The default content type is set to be `application/json`.
+
+  You can also customize the generated `requestBody` specification in 3 ways:
+
+  * add fields `description` and `required`
```

Comment:
I'd like to see a more succinct organization of use cases of `requestBody`. Quoting from #1064 for inspiration:
> single content-type

    schema automatically generated with application/json
        with or without optional properties
    schema automatically generated with explicitly set content-type
        with or without optional properties
    schema set by user

Reply:
@shimks Sorry I don't quite agree with you here, the simplest example and 3 customized scenarios covered the following use cases:

```
schema automatically generated with application/json
    with or without optional properties
schema automatically generated with explicitly set content-type
    with or without optional properties
schema set by user
```

IMO there are three items can be customized:
- optional properties('description', 'required')
- content type
- schema spec

And that's how I classify the use cases.